app/services/foreman/plugin.rb

```diff
@@ -93,6 +93,17 @@ def installed?(id)
 
     def initialize(id)
       @id = id.to_sym
+      add_loggers
+    end
+
+    def add_loggers
+      return unless (defined?(Foreman::Logging) && SETTINGS[@id])
+
+      if (loggers = SETTINGS[@id][:loggers])
```

Comment:
I'm referring to default configuration for logging in plugins, as that's what this seems to depend upon.  It seems that if the settings file is older or not present then we won't have any good defaults for plugin logging as their loggers won't have been configured.

I do agree about keeping Foreman core's logging in a separate config file, that seems like a good idea, but it does mean one rule for core and one for plugins, which isn't good for consistency.

Perhaps break the configuration into two, with the configuration of stuff like paths and rotation in a logging.yaml as that applies to both core+plugins, the per-logger levels in settings files and ensure that both core+plugins can set sensible defaults for their loggers in code without requiring those settings files.

(Sorry if this is a bit rambly.)